aten/src/ATen/core/dispatch/DispatchTable.h

```diff
@@ -243,7 +243,19 @@ class DispatchTable final {
               "Available functions are ", listAllDispatchKeys())
       }
 
-      const std::string dispatch_key_str = dispatch_key.has_value() ? toString(*dispatch_key) : "None";
+      if (dispatch_key.value() == TensorTypeId::QuantizedCPUTensorId) {
+        // If the input is quantized, but the quantization is not supported.
+        TORCH_CHECK(false, "Tried runninig '", operator_name_, "', but it",
+                    " expects non-quantized input. Please, check the",
+                    " documentation for supported kernels.");
+      }
+      // If the input is not quantized, but the kernel is.
+      TORCH_CHECK(!kernels_.lookup(TensorTypeId::QuantizedCPUTensorId), "Tried",
```

Comment:
For consistency, can we pull this conditional out into an `if` statement to make it clear this is not the fall-through path? I was confused when first reading this